streaming/index.js

```diff
@@ -554,7 +628,19 @@ const startServer = async () => {
       return;
     }
 
-    accountFromRequest(req).then(() => checkScopes(req, channelNameFromPath(req))).then(() => {
+    const channelName = channelNameFromPath(req);
+
+    // If no channelName can be found for the request, then we should terminate
+    // the connection, as there's nothing to stream back
+    if (!channelName) {
+      const err = new Error('Unknown channel requested');
+      err.status = 400;
```

Comment:
I've gone with 400 here, instead of 404, though arguably it could be either.

Reply:
404 is used for non-existing endpoints. 400 is generic. What about 410 (Gone)?

Reply:
It's not technically gone, it kind of never existed — basically you could try connecting to `/api/v1/streaming/foo` and that'd hit here, but only because "foo" is an unknown channel. I went with 400 because 404 didn't really feel like it made sense.

Reply:
400 seems fine to me.